datastore/db/bytebase/FertiScan/delete_inspection_function.sql

```diff
@@ -0,0 +1,125 @@
+-- Function to delete organization_information
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".delete_organization_information(
+    p_organization_information_id uuid
+)
+RETURNS void
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    DELETE FROM "fertiscan_0.0.13".organization_information
+    WHERE id = p_organization_information_id;
+END;
+$$;
+
+-- Trigger function to handle after organization_information delete for location deletion
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig()
+RETURNS TRIGGER
+LANGUAGE plpgsql
+AS $$
+BEGIN
+    IF OLD.location_id IS NOT NULL THEN
+        DELETE FROM "fertiscan_0.0.13".location
+        WHERE id = OLD.location_id;
+    END IF;
+
+    RETURN NULL;
+END;
+$$;
+
+-- Trigger definition on organization_information table for location deletion
+DROP TRIGGER IF EXISTS after_organization_information_delete_location ON "fertiscan_0.0.13".organization_information;
+CREATE TRIGGER after_organization_information_delete_location
+AFTER DELETE ON "fertiscan_0.0.13".organization_information
+FOR EACH ROW
+EXECUTE FUNCTION "fertiscan_0.0.13".after_org_info_delete_location_trig();
+
+-- Function to delete an inspection and related data
+CREATE OR REPLACE FUNCTION "fertiscan_0.0.13".delete_inspection(
+    p_inspection_id uuid,
+    p_inspector_id uuid
+)
+RETURNS jsonb
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    inspection_record jsonb;
+BEGIN
+    IF NOT EXISTS (
+        SELECT 1
+        FROM "fertiscan_0.0.13".inspection 
+        WHERE id = p_inspection_id 
+        AND inspector_id = p_inspector_id
+    ) THEN
+        RAISE EXCEPTION 'Inspector % is not the creator of inspection %', p_inspector_id, p_inspection_id;
+    END IF;
+
+    -- Delete the inspection record and retrieve it
+    WITH deleted_inspection AS (
+        DELETE FROM "fertiscan_0.0.13".inspection
+        WHERE id = p_inspection_id
+        RETURNING *
+    )
+    SELECT row_to_json(deleted_inspection) 
```

Comment:
Why do we need to return a json of the deleted row?

Reply:
I return the inspection because I use the picture_set_id it contains afterwards to delete the corresponding picture_set (using existing datastore methods). I could of course:
- return the picture_set_id that would be strange given that the centre of attention is the inspection
- fetch the picture_set_id prior to deleting the inspection, but that's one more unnecessary call to the db

Please, suggest an alternative you find this inappropriate.

Reply:
On Nachet we chose:

>fetch the picture_set_id prior to deleting the inspection, but that's one more unnecessary call to the db

Your solution, 
>return the picture_set_id, but that would be strange given that the centre of attention is the inspection

is also viable and require less db transactions. I was scared this data would be sent to other Layers but your answer satisfy me!

Reply:
you can resolve the comment once read